SnookR/sublist/views.py

```diff
@@ -1,28 +1,16 @@
-from django.shortcuts import render
+from django.shortcuts import render, get_object_or_404
 from django.http import HttpResponse
+from django.views.generic.base import TemplateView
+from sublist.models import Sublist
 
- 
-def wichita(request):
-	return HttpResponse("Wichita's sublist")
 
-def river_roadhouse(request):
-	return HttpResponse("River Roadhouse's sublist")
-
-def mcanulty_and_barrys(request):
-	return HttpResponse("McAnulty and Barry's sublist")
-
-def local66(request):
-	return HttpResponse("Local 66 sublist")
-
-def watertrough(request):
-	return HttpResponse("Watertrough sublist")
-
-def fortune_star(request):
-	return HttpResponse("Fortune Star sublist")
-
-def pub181(request):
-	return HttpResponse("Pub 181 sublist")
-
-def outer_eastside(request):
-	return HttpResponse("Outer Eastside sublist")
+class SublistView(TemplateView):
+    """A view for presenting the sublist template"""
+    template_name = 'sublist/sublist.html'
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        print('kwargs', kwargs)
+        print('Sublist.objects.all()', Sublist.objects.all())
+        context['sublist'] = get_object_or_404(Sublist, slug=kwargs.get('sublist'))
+        return context
```

Comment:
A TemplateView is used for easily presenting a simple template with some data.  `get_context_data` takes `kwargs`, which are the URL keywords, see urls.py to see the regex for the `'sublist'` keyword (hence `kwargs.get('sublist')`) (which is any number of words separated by a hyphen).  

You always get the context from the parent class, and then populate it with the data you want for the template.  In this case we select a Sublist instance who's slug is equal to the 'sublist' keyword from the URL kwargs.  The slug is auto-saved as the `name` field of the Sublist model but lower-cased and with spaces separated by hyphens (for nice urls).  For example, McAnulty and Barrys will have a slug `mcanulty-and-barrys` and the URL will be at `www.thewebsite.com/mcanulty-and-barrys`.

If you look in [sublist.html](https://github.com/bobbyesh/SnookR/blob/13e4b5678c83a00bbc5803713ef7e2c3512be9bb/SnookR/sublist/templates/sublist/sublist.html) you can see how the template has access to a variable called `sublist` that also has a name attribute.  This is the Sublist model instance we get from [this line](https://github.com/eSmelser/SnookR/pull/4/files/13e4b5678c83a00bbc5803713ef7e2c3512be9bb#diff-e0b9f00d28323bfd0e68f0c009c4e0f1R15), the `name` attribute is the name field.

I'm not sure if I explained that well, let me know if you have any questions!